crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -194,7 +196,7 @@ pub(super) async fn configure_dsn(
                         .iter()
                         .filter_map(|rb| {
                             if rb.is_none() {
-                                warn!("Received empty optional root block!");
+                                error!("Received empty optional root block!");
```

Comment:
What I meant by error here is that peer shouldn't respond like this, they must ensure to return either indexes or none, as discussed. So yes, printing error is necessary, but it will be an error due to not having root blocks on the node (I believe we already have cache for it BTW, or is it just for record roots?).

I'm actually thinking now that we should probably change node API to also return either all record roots or none, I can't recall off top of my head why it was necessary to support options there, but I think it may not be necessary anymore.